src/app/app.component.ts

```diff
@@ -17,10 +17,18 @@ export class AppComponent implements OnInit {
   loading!: Observable<boolean>;
 
   constructor(
-    private loadingService: LoadingService
+    private loadingService: LoadingService,
+    private authService: AuthenticationService
   ) {}
+
   
   ngOnInit(): void {
     this.loading = this.loadingService.loadingSubject.asObservable();
+
+    this.authService.getUser()
+      .subscribe(res => {
+        this.authService.userSubject.next(JSON.parse(JSON.stringify(res)));
```

Comment:
All such actions with subject are recommended to be done in the service

Reply:
ok, refactored it with private field of userSubject and using the setter here (and the getter in header component)